### DISCUSSION TITLE
Argocd CLI- "Failed to invoke grpc call"

### DISCUSSION THREAD

**Discussion by Wyifei** [argoproj/argo-cd#8806] [Q&A]
Our company's certificate is terminated at F5 load balancer and F5 connect Nginx ingress controller via Http protocal.
The certificate include domain: argocd-cons.xxx.com

I've add --insecure in argo-server parameter as ingress set as below:

apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: argocd-server-ingress
  namespace: argocd
  annotations:
spec:
  rules:
  - host: argocd-cons.xxx.com
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: argocd-server
            port:
              name: https

Currently, I could login argocd Web UI via https://argocd-cons.xxx.com, but when I try to use argocd CLI, system prompt:
WARN[0002] Failed to invoke grpc call. Use flag --grpc-web in grpc calls. To avoid this warning message, use flag --grpc-web

How could I avoid this warning message?  By the way, we must terminate SSL in F5 according to our policy.

**Comment by dongho-jung** [comment]
I'm suffering from this too D:

**Comment by JSNortal** [comment]
Same issue for myself.

**Comment by zMynxx** [comment]
Just redirect this warning to a blackhole if it bothers you, e.g. `2>/dev/null` at the end of the command.

**Comment by jlucktay** [comment]
Is there a config setting or an environment variable or some such where we could set the `--grpc-web` flag to be true all the time?
This way, we wouldn't have to provide it for every `argocd` call from the command line.

**Reply by jlucktay** [reply]
Went spelunking through the code and found my way back [to this](https://argo-cd.readthedocs.io/en/stable/user-guide/environment-variables/).
**tl;dr:** `export ARGOCD_OPTS="--grpc-web"` in your shell.

**Reply by thiagowfx** [reply]
See https://github.com/argoproj/argo-cd/discussions/8806#discussioncomment-9569142

**Comment by thiagowfx** [comment]
@jlucktay https://stackoverflow.com/questions/78503399/how-to-set-grpc-web-by-default-for-argocd-invocations:

Pass `--grpc-web` to `argocd login`. The config should persist in `~/.config/argocd/config`. Tested, worked for me.
